server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java

```diff
@@ -64,27 +64,26 @@ public void handle(Request request, Response response) throws Exception {
     response.noContent();
   }
 
-  private void doHandle(String idParam) {
+  private void doHandle(String id) {
     try (DbSession dbSession = dbClient.openSession(false)) {
-      long id = Long.parseLong(idParam);
-      ComponentLinkDto link = dbClient.componentLinkDao().selectById(dbSession, id);
+      ProjectLinkDto link = dbClient.projectLinkDao().selectByUuid(dbSession, id);
 
       link = WsUtils.checkFound(link, "Link with id '%s' not found", id);
       checkProjectAdminPermission(link);
       checkNotProvided(link);
 
-      dbClient.componentLinkDao().delete(dbSession, link.getId());
+      dbClient.projectLinkDao().delete(dbSession, link.getUuid());
       dbSession.commit();
     }
   }
 
-  private static void checkNotProvided(ComponentLinkDto link) {
+  private static void checkNotProvided(ProjectLinkDto link) {
     String type = link.getType();
     boolean isProvided = type != null && PROVIDED_TYPES.contains(type);
     WsUtils.checkRequest(!isProvided, "Provided link cannot be deleted.");
   }
 
-  private void checkProjectAdminPermission(ComponentLinkDto link) {
-    userSession.checkComponentUuidPermission(UserRole.ADMIN, link.getComponentUuid());
+  private void checkProjectAdminPermission(ProjectLinkDto link) {
+    userSession.checkComponentUuidPermission(UserRole.ADMIN, link.getProjectUuid());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "checkComponentUuidPermission"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fproject-links&issues=AWGTabq5QeDOAQ7I7d6_&open=AWGTabq5QeDOAQ7I7d6_)

[ISSUE_KEY]: AWGTabq5QeDOAQ7I7d6_